keps/sig-node/5573-cgroup-v1-unsupported/README.md

```diff
@@ -0,0 +1,510 @@
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [x] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [x] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Remove cgroup v1 support from Kubernetes codebase, building upon the maintenance mode introduced in KEP-4569. This enhancement will remove cgroup v1 support.
+
+This KEP will have a phased approach where beta will prevent kubelet from starting on a cgroup v1 node.
+
+In n+3 releases, we will remove the cgroup v1 code.
+
+## Motivation
+
+Following the transition of cgroup v1 support to maintenance mode in KEP-4569, the next logical step is to move cgroup v1 to an unsupported state. This aligns with the broader ecosystem's migration to cgroup v2, including major Linux distributions and the Linux kernel community's focus on cgroup v2 for new features and improvements.
+
+The motivation builds on the rationale established in KEP-4569:
+- The Linux kernel community has made cgroup v2 the focus for new features
+- Major Linux distributions are phasing out cgroup v1 support
```

Comment:
Do we have a more detailed timeline for this? Because any hosts on cgroup v1 running `minikube --vm-driver=docker`, `kind`,  `k3d` etc will be inheriting cgroup v1/v2 from the host.

Many of our users will not be on bleeding edge distros, so even when systemd drops support in the latest release they may still be on cgroup v1.

Reply:
For me, it will be easier to focus on cgroupv2, but I still commonly see users on v1 ...

Reply:
I can only dig up fedora ecosystem as I know that better.

RHEL 9.4 has deprecated cgroup v1 and RHEL 10 has removed cgroup for cgroup v1.

in Openshift we have deprecated cgroup v1 in 4.18 and block upgrades to 4.19 (1.31 to 1.32).

Reply:
> What does "certified" means in this context?

Sorry, very confusing and overloaded terms here -- I mean conformant Kubernetes distros (not linux distros) which in turn are not using systemd, I know at least one fairly popular off the top of my head, I didn't want to focus the conversation on that though, I just mean that systemd is not universal in supported clusters.

Reply:
I added a section on linus distributions.